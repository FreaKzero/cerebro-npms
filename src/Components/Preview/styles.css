.main {
}

.cardFooter {
  border-top: 1px solid #dbdbdb;
  -webkit-box-align: stretch;
      -ms-flex-align: stretch;
          align-items: stretch;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.cardFooterItem {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-preferred-size: 0;
      flex-basis: 0;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  -ms-flex-negative: 0;
      flex-shrink: 0;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  padding: 0.75rem;
}

.cardFooterItem:not(:last-child) {
  border-right: 1px solid #dbdbdb;
}

.card {
  background-color: white;
  -webkit-box-shadow: 0 2px 3px rgba(10, 10, 10, 0.1), 0 0 0 1px rgba(10, 10, 10, 0.1);
          box-shadow: 0 2px 3px rgba(10, 10, 10, 0.1), 0 0 0 1px rgba(10, 10, 10, 0.1);
  color: #4a4a4a;
  max-width: 100%;
  position: relative;
  border-radius: 3px;
}

.cardHeader {
  -webkit-box-align: stretch;
      -ms-flex-align: stretch;
          align-items: stretch;
  -webkit-box-shadow: 0 1px 2px rgba(10, 10, 10, 0.1);
          box-shadow: 0 1px 2px rgba(10, 10, 10, 0.1);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.cardHeaderTitle {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  color: #363636;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  font-weight: 700;
  padding: 0.75rem;
}

.cardContent {
  padding: 1.5rem;
}

.version {
  font-size: 12px;
  font-weight: normal;
  font-style: italic;
  color: #616161;
}

.author {
  font-size: 12px;
  font-weight: normal;
  font-style: italic;
}

.headline {
  display: inline-flex;
  width: 100%;
}

.headline .left, .headline .right {
  flex: 1;
}

.headline .right {
  text-align: right;
  max-width: 30%;
}

.markdown p {
  margin: 0 0 10px 0;
  padding: 0;
}