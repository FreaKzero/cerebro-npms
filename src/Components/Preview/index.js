import React, { Component, PropTypes } from 'react';
import Scores from '../Scores';
import Flags from '../Flags';
import Links from '../Links';
import Tags from '../Tags';
import ReactMarkdown from 'react-markdown';
import styles from './styles.css'

export default ({ item, actions }) => (
  <div className={styles.main}>
    <div className='card' tabIndex='1'>
      
      <div className='card-header'>
        <span className='card-header-title'>
          <div className={styles.headline}>
            <div className={styles.left}>
              <h3>{item.package.name}</h3> <span className={styles.version}>({item.package.version})</span>
              <br />
              <a 
                className={styles.author} 
                onClick={() => actions.open(`https://www.npmjs.com/~${item.package.publisher.username}`)}
              >
                {item.package.publisher.username} &lt;{item.package.publisher.email}&gt;
              </a>
            </div>
            <div className={styles.right}>
              <Flags 
                flags={item.flags || {}} 
                size='big' 
              />
            </div>
          </div>
        </span> 
      </div>

      <div className='card-content'>
        <div className='content'>
          <div className={styles.markdown}>
            {item.package.description ? 
              <ReactMarkdown source={item.package.description} />
              :
              <p>No Description given</p>
            }
            
          </div>
          <Scores
            popularity={Math.trunc(item.score.detail.popularity * 100)}
            maintenance={Math.trunc(item.score.detail.maintenance * 100)}
            quality={Math.trunc(item.score.detail.quality * 100)}
          />
          <Tags item={item} />
        </div>
      </div>
      <Links item={item} actions={actions} />
    </div>
  </div>
)