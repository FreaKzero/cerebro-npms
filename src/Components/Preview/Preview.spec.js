import React from 'react';
import Preview from './index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const item =
    {
    package: {
    name: "dlvr",
    scope: "unscoped",
    version: "4.0.5",
    description: "DeLiVeR - Command Line Tool to automate releasing processes on GitHub/GitLab/npm",
    keywords: [
    "npm",
    "release",
    "changelog",
    "cli",
    "gitlab",
    "management"
    ],
    date: "2017-11-25T22:48:02.614Z",
    links: {
    npm: "https://www.npmjs.com/package/dlvr",
    homepage: "https://github.com/freakzero/dlvr#readme",
    repository: "https://github.com/freakzero/dlvr",
    bugs: "https://github.com/freakzero/dlvr/issues"
    },
    author: {
    name: "FreaKzero"
    },
    publisher: {
    username: "freakzero",
    email: "pete@freakzero.com"
    },
    maintainers: [
    {
    username: "freakzero",
    email: "pete@freakzero.com"
    }
    ]
    },
    score: {
    final: 0.6529780775444187,
    detail: {
    quality: 0.9392171854054061,
    popularity: 0.06060838884010745,
    maintenance: 0.9999999595107407
    }
    },
    searchScore: 100000.56
    };
    
    
    const actions = {
      open: () => { return; }
    }

  const tree = renderer
    .create(<Preview item={item} actions={actions} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});