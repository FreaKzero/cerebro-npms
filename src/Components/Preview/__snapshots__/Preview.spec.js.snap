// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={undefined}
>
  <div
    className="card"
    tabIndex="1"
  >
    <div
      className="card-header"
    >
      <span
        className="card-header-title"
      >
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            <h3>
              dlvr
            </h3>
             
            <span
              className={undefined}
            >
              (
              4.0.5
              )
            </span>
            <br />
            <a
              className={undefined}
              onClick={[Function]}
            >
              freakzero
               &lt;
              pete@freakzero.com
              &gt;
            </a>
          </div>
          <div
            className={undefined}
          >
            <span
              className={undefined}
            >
              <span
                className={undefined}
              />
            </span>
          </div>
        </div>
      </span>
    </div>
    <div
      className="card-content"
    >
      <div
        className="content"
      >
        <div
          className={undefined}
        >
          <div>
            <p>
              DeLiVeR - Command Line Tool to automate releasing processes on GitHub/GitLab/npm
            </p>
          </div>
        </div>
        <div
          className={undefined}
        >
          <div>
            <p>
              Quality
            </p>
            <h4
              style={
                Object {
                  "color": "#77ab59",
                }
              }
            >
              93
              %
            </h4>
            <svg
              className="rc-progress-circle "
              style={Object {}}
              viewBox="0 0 100 100"
            >
              <path
                className="rc-progress-circle-trail"
                d="M 50,50 m 0,-45
                   a 45,45 0 1 1 0,90
                   a 45,45 0 1 1 0,-90"
                fillOpacity="0"
                stroke="#D9D9D9"
                strokeWidth={1}
                style={
                  Object {
                    "strokeDasharray": "282.7433388230814px 282.7433388230814px",
                    "strokeDashoffset": "-0px",
                    "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
                  }
                }
              />
              <path
                className="rc-progress-circle-path"
                d="M 50,50 m 0,-45
                   a 45,45 0 1 1 0,90
                   a 45,45 0 1 1 0,-90"
                fillOpacity="0"
                stroke="#77ab59"
                strokeLinecap="butt"
                strokeWidth="10"
                style={
                  Object {
                    "strokeDasharray": "262.9513051054657px 282.7433388230814px",
                    "strokeDashoffset": "-0px",
                    "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                  }
                }
              />
            </svg>
          </div>
          <div>
            <p>
              Maintenance
            </p>
            <h4
              style={
                Object {
                  "color": "#77ab59",
                }
              }
            >
              99
              %
            </h4>
            <svg
              className="rc-progress-circle "
              style={Object {}}
              viewBox="0 0 100 100"
            >
              <path
                className="rc-progress-circle-trail"
                d="M 50,50 m 0,-45
                   a 45,45 0 1 1 0,90
                   a 45,45 0 1 1 0,-90"
                fillOpacity="0"
                stroke="#D9D9D9"
                strokeWidth={1}
                style={
                  Object {
                    "strokeDasharray": "282.7433388230814px 282.7433388230814px",
                    "strokeDashoffset": "-0px",
                    "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
                  }
                }
              />
              <path
                className="rc-progress-circle-path"
                d="M 50,50 m 0,-45
                   a 45,45 0 1 1 0,90
                   a 45,45 0 1 1 0,-90"
                fillOpacity="0"
                stroke="#77ab59"
                strokeLinecap="butt"
                strokeWidth="10"
                style={
                  Object {
                    "strokeDasharray": "279.9159054348506px 282.7433388230814px",
                    "strokeDashoffset": "-0px",
                    "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                  }
                }
              />
            </svg>
          </div>
          <div>
            <p>
              Popularity
            </p>
            <h4
              style={
                Object {
                  "color": "#d41243",
                }
              }
            >
              6
              %
            </h4>
            <svg
              className="rc-progress-circle "
              style={Object {}}
              viewBox="0 0 100 100"
            >
              <path
                className="rc-progress-circle-trail"
                d="M 50,50 m 0,-45
                   a 45,45 0 1 1 0,90
                   a 45,45 0 1 1 0,-90"
                fillOpacity="0"
                stroke="#D9D9D9"
                strokeWidth={1}
                style={
                  Object {
                    "strokeDasharray": "282.7433388230814px 282.7433388230814px",
                    "strokeDashoffset": "-0px",
                    "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
                  }
                }
              />
              <path
                className="rc-progress-circle-path"
                d="M 50,50 m 0,-45
                   a 45,45 0 1 1 0,90
                   a 45,45 0 1 1 0,-90"
                fillOpacity="0"
                stroke="#d41243"
                strokeLinecap="butt"
                strokeWidth="10"
                style={
                  Object {
                    "strokeDasharray": "16.964600329384883px 282.7433388230814px",
                    "strokeDashoffset": "-0px",
                    "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
                  }
                }
              />
            </svg>
          </div>
        </div>
        <div
          className={undefined}
        >
          <span
            className="tag"
          >
            <img
              className={undefined}
              src="tag.png"
            />
             
            npm
          </span>
          <span
            className="tag"
          >
            <img
              className={undefined}
              src="tag.png"
            />
             
            release
          </span>
          <span
            className="tag"
          >
            <img
              className={undefined}
              src="tag.png"
            />
             
            changelog
          </span>
          <span
            className="tag"
          >
            <img
              className={undefined}
              src="tag.png"
            />
             
            cli
          </span>
          <span
            className="tag"
          >
            <img
              className={undefined}
              src="tag.png"
            />
             
            gitlab
          </span>
          <span
            className="tag"
          >
            <img
              className={undefined}
              src="tag.png"
            />
             
            management
          </span>
        </div>
      </div>
    </div>
    <footer
      className="card-footer"
    >
      <div
        className="undefined card-footer-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        tabIndex={0}
      >
        <a>
          npm
        </a>
      </div>
      <div
        className="undefined card-footer-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        tabIndex={0}
      >
        <a>
          homepage
        </a>
      </div>
      <div
        className="undefined card-footer-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        tabIndex={0}
      >
        <a>
          repository
        </a>
      </div>
      <div
        className="undefined card-footer-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        tabIndex={0}
      >
        <a>
          bugs
        </a>
      </div>
    </footer>
  </div>
</div>
`;
